### PR TITLE
Add methods for yaw, pitch, and step height

### PR DESCRIPTION
**Pull request by datatags**
## Info
This PR adds some new methods and fixes a bug.

## Details
I've added getters and setters for three more properties, yaw, pitch, and step height. Yaw was sort of available with `setBodyRotation` and `setHeadRotation` but base entity yaw is treated differently. Pitch and step height are new additions, all added under EntityBody because that seemed like a good place.

Also, this PR adds a null check to the `getEntityTypes` method, because `EntityType.UNKNOWN` has a null class.

Lastly, I'm wondering about having a method that calls `Entity#move(MoverType, Vec3)` because it's useful for causing smooth entity movement. It seems like using `MoverType.SELF` would work for most cases, so I don't think we'd necessarily need a wrapper for `MoverType`, but I'd like to hear what you think about this? (The difference between `Entity#move` and the existing `moveTo` is that `Entity#move` operates at a lower level, and changes the location instantly instead of using the entity's AI to do it.)

### OS
OS: Debian 11

#### MC Builds
- [ ] Tested on Latest Spigot Version
- [X] Tested on Latest Paper / Purpur Version

#### JDK Builds
Tested on:
- [ ] JDK Version 8/11
- [X] JDK Version 17/18

## Demonstration
Pitch modification test
eSYUOxFp9H

### CODE REVIEW

File: 1_18_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R2.java [hunk 1]
```diff
@@ -1629,7 +1659,9 @@ private static EntityType[] getEntityTypes(Class<?>... nms) {
         for (Class<?> c : nms) {
 
             Class<? extends Entity> bukkit = fromNMS((Class<? extends net.minecraft.world.entity.Entity>) c, Entity.class);
-            for (EntityType t : EntityType.values()) if (t.getEntityClass().isAssignableFrom(bukkit)) types.add(t);
+            for (EntityType t : EntityType.values()) {
```
Comment (GamerCoder215) [inline 1_18_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R2.java:1662]: Didn't notice that, thanks though.

**Review by GamerCoder215 — APPROVED** [review]
LGTM!

**Comment by GamerCoder215** [comment]
We'll have to wrap all classes to work with older versions of NMS (below 1.17) that have the version packages in their name.

Adding separate MoveTypes to the EntityController would be a good idea if you think that method performs better. Alternatively, I could scrap using the controllers entirely and convert to that method.

I'll go ahead and merge this, for now, I'll start working on a draft for this when I get the chance. Thanks for pointing it out!

**Comment by datatags** [comment]
> We'll have to wrap all classes to work with older versions of NMS (below 1.17) that have the version packages in their name.
> 
> Adding separate MoveTypes to the EntityController would be a good idea if you think that method performs better. Alternatively, I could scrap using the controllers entirely and convert to that method.
> 
> I'll go ahead and merge this, for now, I'll start working on a draft for this when I get the chance. Thanks for pointing it out!

I think it'd be good to have both, they serve different purposes. `moveTo`, as you know, makes an entity find a path to a block and go there, however long it takes. The `Entity#move` method is more like a teleport, it's the part that does the actual position change for the entity each tick.

I'm just unsure if it'd be worth the effort to wrap the `MoveType` enum since in almost all cases, `MoveType.SELF` is used. The options are `SELF`, `PLAYER`, `PISTON`, `SHULKER_BOX`, and `SHULKER`.

Thanks!

**Comment by GamerCoder215** [comment]
> > We'll have to wrap all classes to work with older versions of NMS (below 1.17) that have the version packages in their name.
> > Adding separate MoveTypes to the EntityController would be a good idea if you think that method performs better. Alternatively, I could scrap using the controllers entirely and convert to that method.
> > I'll go ahead and merge this, for now, I'll start working on a draft for this when I get the chance. Thanks for pointing it out!
> 
> I think it'd be good to have both, they serve different purposes. `moveTo`, as you know, makes an entity find a path to a block and go there, however long it takes. The `Entity#move` method is more like a teleport, it's the part that does the actual position change for the entity each tick.
> 
> I'm just unsure if it'd be worth the effort to wrap the `MoveType` enum since in almost all cases, `MoveType.SELF` is used. The options are `SELF`, `PLAYER`, `PISTON`, `SHULKER_BOX`, and `SHULKER`.
> 
> Thanks!

Taking a quick look, it seems that the different moving types represent different natural moving methods for entities- like how you specified with the smooth move animation.

[`2ff0b85`](https://github.com/GamerCoder215/MobChip/commit/2ff0b85) has a `naturalMoveTo` movement that should be included in 1.6.0. Thanks for the suggestion!

**Pull request merged**